packages/firestore/src/local/shared_client_state.ts

```diff
@@ -752,13 +757,11 @@ export class WebStorageSharedClientState implements SharedClientState {
   }
 
   shutdown(): void {
-    assert(
-      this.started,
-      'WebStorageSharedClientState.shutdown() called when not started'
-    );
-    this.platform.window.removeEventListener('storage', this.storageListener);
-    this.removeItem(this.localClientStorageKey);
-    this.started = false;
+    if (this.started) {
+      this.platform.window.removeEventListener('storage', this.storageListener);
```

Comment:
Should we remove the 'unload' hook too?  (FWIW that might mean we don't need the `this.started` check, though I don't know if that's guaranteed or not... so probably fine to leave it.)

Reply:
I'm tempted not to. It provides very little gain and adds more state keeping (I would have to keep a reference to the original function pointer for it to work consistently in all browsers).